File: esp32_robo_arm/python/test_websocket.py
#!/usr/bin/env python3
"""
Тест WebSocket соединения
"""

import asyncio
import websockets
import json

async def test_websocket():
    """Тест WebSocket соединения"""
    try:
        uri = "ws://localhost:8000/ws"
        print(f"Подключение к {uri}...")
        
        async with websockets.connect(uri) as websocket:
            print("✅ WebSocket подключен успешно!")
            
            # Отправляем команду получения статуса
            await websocket.send(json.dumps({"command": "get_status"}))
            print("📤 Отправлена команда get_status")
            
            # Получаем ответ
            response = await websocket.recv()
            data = json.loads(response)
            print(f"📥 Получен ответ: {data}")
            
            # Отправляем команду подключения к роботу
            await websocket.send(json.dumps({"command": "connect"}))
            print("📤 Отправлена команда connect")
            
            # Получаем ответ
            response = await websocket.recv()
            data = json.loads(response)
            print(f"📥 Получен ответ: {data}")
            
    except Exception as e:
        print(f"❌ Ошибка WebSocket: {e}")

if __name__ == "__main__":
    asyncio.run(test_websocket())

File: esp32_robo_arm/test_websocket.js
#!/usr/bin/env node

const WebSocket = require('ws');

const ws = new WebSocket('ws://localhost:8000/ws');

ws.on('open', function open() {
  console.log('WebSocket connected');
  
  // Тестируем команду get_status
  ws.send(JSON.stringify({command: 'get_status'}));
});

ws.on('message', function message(data) {
  console.log('Received:', data.toString());
  
  // Тестируем команду connect
  setTimeout(() => {
    ws.send(JSON.stringify({command: 'connect'}));
  }, 1000);
  
  // Тестируем команду move_motor
  setTimeout(() => {
    ws.send(JSON.stringify({
      command: 'move_motor',
      motor: 0,
      direction: 'forward',
      speed: 200
    }));
  }, 2000);
  
  // Закрываем соединение
  setTimeout(() => {
    ws.close();
  }, 3000);
});

ws.on('error', function error(err) {
  console.error('WebSocket error:', err);
});

ws.on('close', function close() {
  console.log('WebSocket disconnected');
});



